app/pods/agenda/agendaitems/agendaitem/documents/route.js

```diff
@@ -23,14 +23,18 @@ export default class DocumentsAgendaitemAgendaitemsAgendaRoute extends Route {
     this.defaultAccessLevel = await this.store.findRecordByUri('access-level', CONSTANTS.ACCESS_LEVELS.INTERN_REGERING);
   }
 
-  setupController(controller) {
+  async setupController(controller) {
     super.setupController(...arguments);
     const agendaitem = this.modelFor('agenda.agendaitems.agendaitem');
     controller.set('agendaitem', agendaitem);
     controller.set('defaultAccessLevel', this.defaultAccessLevel);
     controller.isEnabledPieceEdit = false;
     controller.isOpenPieceUploadModal = false;
     controller.isOpenPublicationModal = false;
+    const agenda = await agendaitem.get('agenda');
+    controller.currentAgenda = agenda;
+    controller.previousAgenda = await agenda.get('previousVersion');
+    await controller.loadNewPieces.perform();
```

Comment:
De `get()` om een relatie van een ember data record op te halen is niet nodig. Dit kan gewoon met dot-notatie `await agendaitem.agenda` (ongeacht of het ember data model al omgezet is naar een native class; in tegenstelling tot `set()`).

De `await` van de relaties zou ik verhuizen naar de `afterModel`-hook zoals voor `defaultAccessLevel`. Dan wordt daarop al gewacht voor de rendering van de template begint.  Op `setupController` wordt niet gewacht. Die `await controller.loadNewPieces.perform();` gaat wel in `setupController` moeten blijven staan omdat je daar de controller nodig hebt.

De `currentAgenda` en `previousAgenda` zou ik op de controller nog definieren als `tracked` property (ofwel zetten met `set()` zoals `defaultAccessLevel`, maar dat is oude Ember syntax)

Reply:
zal dit morgen bekijken

Reply:
@erikap die task aanroepen in de controller werk ook zonder de await (zelfs bij 50+ documenten krijg je bijna onmiddelijk de groene icons te zien.
Ik heb ook ineens de reeds bestaande controller.set() aangepast.